Document STATE_PATH and persistent state for Elastic Agent on Docker (#6504)
elastic/docs-content: reference/fleet/elastic-agent-container.md

# Run Elastic Agent in a container [elastic-agent-container]

> You can run {{agent}} inside a container — either with {{fleet-server}} or standalone. Docker images for all versions of {{agent}} are available from the [Elastic Docker registry](https://www.docker.elastic.co/r/elastic-agent/elastic-agent). If you are running in {{k8s}}, refer to [run {{agent}} on ECK](/deploy-manage/deploy/cloud-on-k8s/standalone-elastic-agent.md).

Reviewer: ```suggestion
You can run {{agent}} inside a container, either with {{fleet-server}} or standalone. Docker images for all versions of {{agent}} are available from the [Elastic Docker registry](https://www.docker.elastic.co/r/elastic-agent/elastic-agent). If you are running in {{k8s}}, refer to [run {{agent}} on ECK](/deploy-manage/deploy/cloud-on-k8s/standalone-elastic-agent.md).
```

* If running _only_ lightweight monitors, use the `elastic-agent` image (and the `wolfi` variant).
* If running _any_ browser monitors, use the `elastic-agent-complete` image.

Run the following command to pull the `elastic-agent-complete` image:

```terminal subs=true
docker pull docker.elastic.co/elastic-agent/elastic-agent-complete:{{version.stack}}
```

To run Synthetics tests using the hardened [Wolfi](https://github.com/wolfi-dev/) image, run:

```terminal subs=true
docker pull docker.elastic.co/elastic-agent/elastic-agent-complete-wolfi:{{version.stack}}
```

## Step 2: Verify the image (optional) [_step_2_optional_verify_the_image]

> We recommend verifying the signatures included with your downloaded Docker images to ensure that the images are valid.

Reviewer: ```suggestion
Verify the signatures included with your downloaded Docker images to ensure that the images are valid.
```

  --user 1000:1000 \ <1>
  --volume "$(pwd)/elastic-agent-state:/state" \
  --env STATE_PATH=/state \
  docker.elastic.co/elastic-agent/elastic-agent:{{version.stack}}
```

1. Set this to the UID:GID that owns the host directory you bind-mounted. `1000:1000` is the default `elastic-agent` user inside the image, but any UID:GID works as long as it matches the directory's ownership.

**Docker Compose example:**

```yaml subs=true
services:
  elastic-agent:
    image: docker.elastic.co/elastic-agent/elastic-agent:{{version.stack}}
    container_name: elastic-agent
    restart: always
    user: "1000:1000" <1>
    environment:
      - FLEET_ENROLLMENT_TOKEN=<enrollment-token>
      - FLEET_ENROLL=1
      - FLEET_URL=<fleet-server-url>
      - STATE_PATH=/state
    volumes:
      - ./elastic-agent-state:/state
```

1. Set this to the UID:GID that owns the host directory you bind-mounted. `1000:1000` is the default `elastic-agent` user inside the image, but any UID:GID works as long as it matches the directory's ownership.

Refer to [Environment variables](/reference/fleet/agent-environment-variables.md) for all available options.

## Logs [_logs]

> Since a container supports only a single version of {{agent}}, logs and state are stored differently than when running an {{agent}} outside of a container. Logs are stored at `/usr/share/elastic-agent/state/data/logs/*`.

Reviewer: ```suggestion
Because a container supports only a single version of {{agent}}, logs and state are stored differently than when running an {{agent}} outside of a container. Logs are stored at `/usr/share/elastic-agent/state/data/logs/*`.
```
